src/main/webapp/app/overview/overview.component.ts

```diff
@@ -1,28 +1,50 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnDestroy } from '@angular/core';
 import { Course, CourseScoreCalculationService, CourseService } from 'app/entities/course';
 import { HttpResponse } from '@angular/common/http';
 import { JhiAlertService } from 'ng-jhipster';
 import { Exercise, ExerciseService } from 'app/entities/exercise';
 import { AccountService } from 'app/core';
-import { TUM_REGEX } from 'app/app.constants';
+import { GuidedTour } from 'app/guided-tour/guided-tour.constants';
+import { GuidedTourService } from 'app/guided-tour/guided-tour.service';
+import { Subscription } from 'rxjs';
 
 @Component({
     selector: 'jhi-overview',
     templateUrl: './overview.component.html',
-    styles: [],
+    styleUrls: [],
 })
-export class OverviewComponent {
+export class OverviewComponent implements OnDestroy {
     public courses: Course[];
     public nextRelevantCourse: Course;
+    public overviewTour: GuidedTour;
+
+    subscription: Subscription;
 
     constructor(
         private courseService: CourseService,
         private exerciseService: ExerciseService,
         private jhiAlertService: JhiAlertService,
         private accountService: AccountService,
         private courseScoreCalculationService: CourseScoreCalculationService,
+        private guidedTourService: GuidedTourService,
     ) {
         this.loadAndFilterCourses();
+
+        setTimeout(() => {
+            if (guidedTourService.guidedTourSettings && guidedTourService.guidedTourSettings.showCourseOverviewTour) {
+                this.startTour();
+            }
+        }, 500);
```

Comment:
Shouldn't it be possible in this case to use a timeout of 0?
This should wait for 1 re-render of the template.

If it needs to be 500 to work, it will cause Bugs sooner or later.

I also don't like about it, that the component that uses the guided tour service needs to know about this intricacy of the guided tour (it should be more plug&play).